src/network/networkd-util.c

```diff
@@ -99,3 +99,44 @@ int config_parse_address_family_boolean_with_kernel(
 
         return 0;
 }
+
+int ip_port_from_string(const char *s, uint16_t *port) {
+        int r;
+
+        assert(s);
+        assert(port);
+
+        r = safe_atou16(s, port);
+        if (r < 0 || *port == 0)
+                return -EINVAL;
+        return 0;
+}
+
+int cidr_from_string(const char *s, int *family, union in_addr_union *addr, unsigned char *prefixlen) {
```

Comment:
i'd call this parse_address_and_prefixlen() I must say.

The kernel exposes this as "address" and "prefixlen" fields, and doesn't use the term "cidr", and I think we should follow suit here...